### DISCUSSION TITLE
Weekly update 2020-11-20

### DISCUSSION THREAD

**Discussion by alexauroradev** [near/nearcore#3638] [EVM]
https://www.youtube.com/watch?v=mBmbgbuD7-E

# Current goal
Betanet release of EVM support

# Status
We're pretty close. PR (#3257) is ready to be merged, waiting for the review of the Core engineers. Features were used to guard EVM (to safely merge in the master). CI passes the branch. All known issues are resolved.

# Weekly update
**Eugene**
- Wrap `near-evm-runner` behind protocol feature EVM. Before this, it created leakages of external dependency, which resulted in problems with the normal master mode with feature turned off. See more [here](https://github.com/near/nearcore/pull/3257/commits/77c763000018db1d1c73cd2ec76c1eda28dc45ae).
- Creation of new EVM accounts is disabled at the moment
- Meta transactions are disabled at the moment

**Alex**
- No updates

**Arto**
- Validation
  - CryptoZombies tests fixed. The problem was introduces in beginning of Nov regression
  - A bunch of tests for new precompiles (9 in Istanbul) created
- JSON-RPC cooperation with ChainSafe <- calls to be scheduled
- Planned to do an extension of CryptoZombies tests
- Planned to do updates on the documentation

**Bo Yao**
- Helping Eugene with fixing CI
- Since we do not need to decode parameters on the Rust side, they can be forwarded to EVM directly. This knowledge resulted in the PR (#3534) on EIP-712 + `ecrecover`. This PR is passing all CI tests, overall looking good for the merge.

**Mike Purvis**
- Minor changes to `near-web3-provider`
- Changes in PetShop example (settings). Some logic will be kept (meta transactions), but to be hided by now. The changes are planned to be merged soon.
- **[Resolved]** There're problems with account creation in Betanet -- need to consult with SRE team.
- [Hardhat](https://hardhat.org/) (ex-Builder) research. Hardhat is trending upgraded version of Truffle. It can have custom `web3-provider` to be plugged in it. We should consider supporting this tool, since many devs will require it.

**Discussion** 
- Alex: Who need to review the EVM code? Max has rejected the PR, so he need to do extra steps
- Mike: Should we remake docs according to feature changes? No
- Arto, Mike: Hardhat prevails in the current market
- Mike: Let's have 2-3 days a month to play with EVM
- Alex: What is the next goal?
  - Benchmarks, tests, some examples
  - Need to have some contracts that can trigger edge cases
  - Maybe we need to get buggy contracts from the history of the Ethereum and run it on our EVM
  - Need to check the gas limits

**Comment by bowenwang1996** [comment]
> There're problems with account creation in Betanet -- need to consult with SRE team.

I believe this is now resolved
